websites/B/Battlefield Portal/presence.ts

```diff
@@ -0,0 +1,110 @@
+const presence = new Presence({
+    clientId: "919182644296683520"
+});
+async function parse_grpc(sessionID : string, playgroundID : string) : Promise<string> {
+    const a = await fetch("https://kingston-prod-wgw-envoy.ops.dice.se/web.communitygames.CommunityGames/getPlayground", {
+        "credentials": "omit",
+        "headers": {
+            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:95.0) Gecko/20100101 Firefox/95.0",
+            "Accept": "*/*",
+            "Accept-Language": "en-US,en;q=0.5",
+            "x-gateway-session-id": sessionID,
+            "x-dice-tenancy": "prod_default-prod_default-kingston-common",
+            "Content-Type": "application/grpc-web+proto",
+            "X-User-Agent": "grpc-web-javascript/0.1",
+            "X-Grpc-Web": "1",
+            "Sec-Fetch-Dest": "empty",
+            "Sec-Fetch-Mode": "cors",
+            "Sec-Fetch-Site": "cross-site"
+        },
+        "referrer": "https://portal.battlefield.com/",
+        "body": "\u0000\u0000\u0000\u0000&\u0012$"+playgroundID,
+        "method": "POST",
+        "mode": "cors"
+    })
+    if(!a.ok){
+        throw new Error("Failed")
+    }
+    return a.text()
+};
+
+let called = false
+let editing_stamp = Math.floor(Date.now() / 1000);
+let playgroundID: string = null
+let info: string[] = null
+
+const presenceData: PresenceData = {
+    largeImageKey: "img1",
+    details : "",
+    state : "",
+    startTimestamp :null
+  };
+
+presence.on("UpdateData", async () => {
+    let sessionID = document.cookie.match(
```

Comment:
Camel case (change in other instances too)
```suggestion
    let sessionId = document.cookie.match(
```